lib/core/src/Cardano/Wallet.hs

```diff
@@ -1570,49 +1621,45 @@ balanceTransaction
         -- transaction considering only the maximum cost, and only after, try to
         -- adjust the change and ExUnits of each redeemer to something more
         -- sensible than the max execution cost.
-        let txPlutusScriptExecutionCost = maxScriptExecutionCost tl pp redeemers
-        let txContext = defaultTransactionCtx
-                { txPlutusScriptExecutionCost
-                , txMetadata
-                , txWithdrawal
-                , txAssetsToMint = (txAssetsToMint ^. #txTokenMap, Map.empty)
-                , txAssetsToBurn = (txAssetsToBurn ^. #txTokenMap, Map.empty)
-                , txCollateralRequirement =
-                    if txPlutusScriptExecutionCost > Coin 0 then
-                        SelectionCollateralRequired
-                    else
-                        SelectionCollateralNotRequired
-                } & padFeeEstimation partialTx
-
-        -- FIXME: The coin selection and reported fees will likely be wrong in
-        -- the presence of certificates (and deposits / refunds). An immediate
-        -- "fix" is to return a proper error from the handler when any key or
-        -- pool registration (resp. deregistration) certificate is found in the
-        -- transaction. A long-term fix is to handle this case properly during
-        -- balancing.
-        let transform s sel =
+
+        randomSeed <- stdGenSeed
+        let
+            transform :: s -> Selection -> ([(TxIn, TxOut)], [TxIn], [TxOut])
+            transform s sel =
                 let (sel', _) = assignChangeAddresses generateChange sel s
                     inputs = F.toList (sel' ^. #inputs)
-                 in ( selectionDelta txOutCoin sel'
-                    , inputs
+                in  ( inputs
                     , fst <$> (sel' ^. #collateral)
                     , sel' ^. #change
                     )
-        withExceptT ErrBalanceTxSelectAssets $
-            selectAssets @_ @m @s @k ctx pp SelectAssetsParams
-                { outputs
-                , pendingTxs
-                , randomSeed = Nothing
-                , txContext
-                , utxoAvailableForInputs
-                , utxoAvailableForCollateral
-                , wallet
-                , selectionStrategy = SelectionStrategyOptimal
-                }
-                transform
+
+        lift $ traceWith tr $ MsgSelectionForBalancingStart
+            (CS.toExternalUTxOMap $ UTxOIndex.toMap internalUtxoAvailable)
```

Comment:
I suspect this conversion can be avoided if we don't require ourselves to include a `UTxO` value in the `MsgSelectionForBalancingStart` value, since we only ever care about the size, which is the same before and after applying `toExternalUTxOMap`.

I'll push a commit that removes this conversion.

Reply:
Fixed in 80f346ce232504e4cf01094193e660e5646793ad.